File: programmers/lv0/코딩기초트레이닝/조건문/등차수열의특정한항만더하기/Solution.java
package lv0.코딩기초트레이닝.조건문.등차수열의특정한항만더하기;

import java.util.stream.IntStream;

class Solution {
    public static int solutiong(int a, int d, boolean[] included) {
        return IntStream.range(0, included.length)
                .filter(idx -> included[idx])
                .map(idx -> a + (idx * d))
                .reduce(0, Integer::sum);
    }
}

File: programmers/lv0/코딩기초트레이닝/조건문/등차수열의특정한항만더하기/Solution.kt
package lv0.코딩기초트레이닝.조건문.등차수열의특정한항만더하기

class Solution {
    // simple
//    fun solution(a: Int, d: Int, included: BooleanArray) = included.indices.filter { included[it] }.sumOf { a + d * it }

    fun solution(a: Int, d: Int, included: BooleanArray): Int {
        return included.foldIndexed(0) { idx, sum, include ->
            if (include) sum + a + (idx * d) else sum
        }
    }

}